app/controllers/notifications_controller.rb

```diff
@@ -1,20 +1,25 @@
 class NotificationsController < ApplicationController
   before_filter :if_not_signed_in
-  before_action :set_notification, only: [:destroy]
 
   # DELETE /notifications/1
   # DELETE /notifications/1.json
   def destroy
-    @notification.destroy
+    notification = Notification.find_by(
+      id: params[:id],
+      userid: current_user.id
+    )
+
+    notification.destroy if notification.present?
+
     respond_to do |format|
-        format.html { redirect_to :back }
+      format.html { redirect_to :back }
       format.json { head :no_content }
     end
   end
 
   def clear
```

Comment:
```
                          clear_notifications GET      /notifications/clear(.:format)               notifications#clear
    fetch_notifications_notifications GET      /notifications/fetch_notifications(.:format) notifications#fetch_notifications
                   signed_in_notifications GET      /notifications/signed_in(.:format)           notifications#signed_in
                                    notifications GET      /notifications(.:format)                     notifications#index
                                      notification DELETE   /notifications/:id(.:format)                 notifications#destroy
```

I don't think the route for `clear` should be using `HTTP GET` since destroying all notifications for a user is a very stateful request.  It's also susceptible to CSRF attacks since a malicious user could give someone a relatively benign looking link like `http://if-me.org/notifications/clear` and the act of clicking on it (provided they were logged in) would delete all of their notifications whether they liked it or not.

http://programmers.stackexchange.com/questions/188860/why-shouldnt-a-get-request-change-data-on-the-server

Mind if I create an issue for switching this over?